README.md

```diff
@@ -153,6 +153,54 @@ const result1 = awilixManager.getWithTags(diContainer, ['queue'])
 const result2 = awilixManager.getWithTags(diContainer, ['queue', 'low-priority'])
 ```
 
+## Fetching dependencies based on a predicate
+
+In some cases you may want to get dependencies based on whether they satisfy some condition.
+You can use `getByPredicate` method for that:
+
+```js
+import { AwilixManager } from 'awilix-manager'
+import { asClass, createContainer } from 'awilix'
+
+const diContainer = createContainer({
+  injectionMode: 'PROXY',
+})
+
+class QueueConsumerHighPriorityClass {
+}
+
+class QueueConsumerLowPriorityClass {
+}
+
+diContainer.register(
+  'dependency1',
+  asClass(QueueConsumerHighPriorityClass, {
+    lifetime: 'SINGLETON',
+    asyncInit: true,
+  }),
+)
+diContainer.register(
+  'dependency2',
+  asClass(QueueConsumerLowPriorityClass, {
+    lifetime: 'SINGLETON',
+    asyncInit: true,
+  }),
+)
+
+const awilixManager = new AwilixManager({
+  diContainer,
+  asyncInit: true,
+  asyncDispose: true,
+})
+
+// This will return dependency1
+const result1 = awilixManager.getByPredicate((entry) => entry instanceof QueueConsumerHighPriorityClass)
```

Comment:
🟢 Just to confirm my understanding, `result1` will be a `Record<String, any>`, right?

Reply:
improved